src/linear_algebra/sesquilinear_form.lean

```diff
@@ -39,62 +40,93 @@ section comm_ring
 
 -- the `ₗ` subscript variables are for special cases about linear (as opposed to semilinear) maps
 variables {R : Type*} {M : Type*} [comm_semiring R] [add_comm_monoid M] [module R M]
-  {I : R →+* R}
+  {I₁ : R →+* R} {I₂ : R→+* R}
```

Comment:
My guess is that we could support having ring homs between different rings (i.e. have `I₁ : R₁ →+* R₃` and `I₂ : R₂ →+* R₃` essentially for free. Likewise, we should support bilinear forms from two different spaces, i.e. `M₁ →ₛₗ[I₁] M₂ →ₛₗ[I₂] R` (in fact I have a book on my desk right now that does exactly this!).

Reply:
changed it. but I am curious, what is that book?

Reply:
"A Course in Convexity" by Alexander Barvinok, in a chapter about linear programming. He calls bilinear forms "dualities" between vector spaces.